.github/scripts/validateImmutableActionRefs.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+############################################
+#    Check for unsafe action references    #
+############################################
+
+SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &> /dev/null && pwd)"
+readonly SCRIPT_DIR
+GITHUB_DIR="$(dirname "$SCRIPT_DIR")"
+readonly GITHUB_DIR
+
+source "$SCRIPT_DIR/shellUtils.sh"
+
+title 'Checking for mutable action references...'
+
+ACTION_USAGES=''
+
+# Find yaml files in the `.github` directory
+# shellcheck disable=SC2044
+for FILE in $(find "$GITHUB_DIR" -type f \( -name "*.yml" -o -name "*.yaml" \)) ; do
+    USES_LINES="$(grep --no-filename 'uses:' "$FILE")"
```

Comment:
NAB why do you need `--no-filename`? Grep doesn't typically print the file names by default